File: jsontest.py
from multiformats import CID
from dag_json import decode, encode
from multiformats import multihash

obj = {
  "x": 1,
  # CID instances are encoded as links 
  "y": [2, 3, CID.decode('QmaozNR7DZHQK1ZcU9p7QdrshMvXqWK6gpu5rmrkPdT3L4')],
  "z": {
    "a": CID.decode('QmaozNR7DZHQK1ZcU9p7QdrshMvXqWK6gpu5rmrkPdT3L4'),
    "b": None,
    "c": 'string'
  }
}

encoded = encode(obj)
digest = multihash.digest(encoded, "sha2-256")
cidv0 = CID("base58btc", 0, "dag-pb", digest)
cidv1 = CID("base32", 1, "raw", digest)

print("obj")
print(obj)
print("\ndigest")
print(digest)
print("\ncidv0")
print(cidv0)
print("\ncidv1")
print(cidv1)

File: jsontest.js
import { encode, decode } from '@ipld/dag-json'
import { CID } from 'multiformats'
import { sha256 } from 'multiformats/hashes/sha2'

import * as dagPB from '@ipld/dag-pb'
import { code } from 'multiformats/codecs/json'

const obj = {
  x: 1,
  /* CID instances are encoded as links */
  y: [2, 3, CID.parse('QmaozNR7DZHQK1ZcU9p7QdrshMvXqWK6gpu5rmrkPdT3L4')],
  z: {
    a: CID.parse('QmaozNR7DZHQK1ZcU9p7QdrshMvXqWK6gpu5rmrkPdT3L4'),
    b: null,
    c: 'string'
  }
}


let encoded = encode(obj)
const hash = await sha256.digest(encoded)
const cidv0 = CID.create(0, dagPB.code, hash)
const cidv1 = CID.create(1, code, hash)



console.log("obj")
console.log(obj)
console.log("\ncidv0")
console.log(cidv0)
console.log("\ncidv1")
console.log(cidv1)

// let decoded = decode(encoded)
// decoded.y[0] // 2
// let test_var = CID.asCID(decoded.z.a) // cid instance
// console.log("\n\ntest_var")
// console.log(test_var)

